### ISSUE TITLE
Entrypoint Selector: It's surprising to see something deploy without explicitly clicking a Deploy button

### ISSUE THREAD

**Issue by kgartland-rstudio** [posit-dev/publisher#2017]
I'm finding that I don't really expect the actual deploy to occur when I'm using the Entrypoint Selector.  It feels like this is a good way to setup your deployment but I think it's missing an actual Deploy button.  

In contrast, during the Home View flow, you enter the multi-stepper and complete all the steps, then it returns you to the Home View to "Deploy Your Project".

When using the Entrypoint Selector, can we instead kick back out to the Home View to allow the customer a chance to make any final considerations before clicking "Deploy Your Project"?

**Comment by sagerb** [comment]
I could see this as a configurable option...

**Comment by dotNomad** [comment]
The other option to avoid some confusion here is to not deploy right away on the creation of a new Deployment so users can read the Configuration to make sure it is valid.

**Comment by dotNomad** [comment]
I believe the closure of #2180 resolves this since we removed the auto deploying.

**Comment by nealrichardson** [comment]
I'm still sad by this. The phrasing of this issue title is also confusing to me: I didn't view this button as an "entrypoint selector" (whatever that means), I viewed it as a Deploy button. The hover text still starts with the verb "Deploy".

**Comment by dotNomad** [comment]
> I didn't view this button as an "entrypoint selector" (whatever that means)

It has to act as a selector if there is more than one deployment that uses the entrypoint (and one isn't already selected). Otherwise we do not have any context for what the user is intending to deploy unless we assume we should deploy all deployments with this entrypoint.

Based on the divergence of feedback that makes me think this is a candidate for our first setting in the extension to allow that button to auto-deploy. So those that want the immediate deployment can use it, and those that do not do not have to. It should arguably start as `false` to avoid any of the security concerns that were brought up by the people who gave us feedback against the auto-deployment.

**Comment by nealrichardson** [comment]
> It should arguably start as `false` to avoid any of the security concerns that were brought up by the people who gave us feedback against the auto-deployment.

Yeah that's fine and totally reasonable. I can still be sad though :)

**Comment by dotNomad** [comment]
> I can still be sad though :)

Hopefully mitigated a bit by a setting at least 😅

**Comment by jonkeane** [comment]
This is a bigger conversation than this issue here, but it does make me wonder: what is the problem we are trying to solve with the publish button / icon on top of the entrypoint files? We should take a step back and answer this, which might help provide more clarity about what it's behavior should be.

I'm being intentionally vague here: I know the history and the reason(s) why we added this in the first place — but want to spark a conversation about what problem it's solving for our users _now_.
